cmd/recorder/config.go

```diff
@@ -0,0 +1,165 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+)
+
+type AVFormat string
+
+const (
+	AVFormatMP4 AVFormat = "mp4"
+)
+
+const (
+	// defaults
+	VideoWidthDefault   = 1920
+	VideoHeightDefault  = 1080
+	VideoRateDefault    = 1500
+	AudioRateDefault    = 64
+	FrameRateDefault    = 30
+	OutputFormatDefault = AVFormatMP4
+
+	// limits
+	VideoWidthMin  = 1280
+	VideoWidthMax  = 3840
+	VideoHeightMin = 720
+	VideoHeightMax = 2160
+	VideoRateMin   = 500
+	VideoRateMax   = 10000
+	AudioRateMin   = 32
+	AudioRateMax   = 320
+	FrameRateMin   = 10
+	FrameRateMax   = 60
+)
+
+type RecorderConfig struct {
+	// input config
+	SiteURL   string
+	CallID    string
+	ThreadID  string
+	AuthToken string
+
+	// output config
+	Width        int
+	Height       int
+	VideoRate    int
+	AudioRate    int
+	FrameRate    int
+	OutputFormat AVFormat
+}
+
+func (cfg RecorderConfig) IsValid() error {
+	if cfg == (RecorderConfig{}) {
+		return fmt.Errorf("config cannot be empty")
+	}
+	if cfg.SiteURL == "" {
+		return fmt.Errorf("SiteURL cannot be empty")
+	}
+	if cfg.CallID == "" {
+		return fmt.Errorf("CallID cannot be empty")
+	}
+	if cfg.ThreadID == "" {
+		return fmt.Errorf("ThreadID cannot be empty")
+	}
```

Comment:
I am not that all in love with having to pass this but until we start persisting the call state/info we need to keep some state here as the call could be over when the processing completes on this side.

Reply:
I don't see why it would be a problem.

Reply:
Not a problem per se but it feels like exposing implementation details.